build/ical.js

```diff
@@ -340,7 +340,42 @@ ICAL.helpers = {
 ICAL.design = (function() {
   'use strict';
 
-  var ICAL_NEWLINE = /\\\\|\\;|\\,|\\[Nn]/g;
+  var FROM_ICAL_NEWLINE = /\\\\|\\;|\\,|\\[Nn]/g;
+  var TO_ICAL_NEWLINE = /\\|;|,|\n/g;
+  var FROM_VCARD_NEWLINE = /\\\\|\\,|\\[Nn]/g;
+  var TO_VCARD_NEWLINE = /\\|,|\n/g;
+
+  var textType = function(fromNewline, toNewline, type) {
```

Comment:
Since this function creates a text type, I think you should name it `createTextType`. Also, no need for variable assignment, you can just use `function createTextType(fromNewline, toNewline, type)`